crates/fuel-gas-price-algorithm/src/v1/tests/update_da_record_data_tests.rs

```diff
@@ -270,3 +228,132 @@ fn update_da_record_data__da_block_updates_projected_total_cost_with_known_and_g
     let expected = new_known_total_cost + guessed_part;
     assert_eq!(actual, expected as u128);
 }
+
+#[test]
+fn update_da_record_data__updates_known_total_cost_if_blocks_are_out_of_order() {
```

Comment:
Maybe it's because it's late, but I don't see where we provide blocks out of order in this test case. Is it because we skip `1` in the `recorded_range`?

Reply:
Could we be even more ezplicit and set `recorded_range` to `[3, 2]` instead?.

Reply:
That's a good idea. I've switched it. In reality, they'll probably be in order inside the vector, but they don't need to be. The point of this PR is that some heights can be skipped and returned to later. These tests skip `1` and do `2` and `3`.